File: bindings/python/examples/how_tos/accounts_and_addresses/list_accounts.py
import os

from dotenv import load_dotenv

from iota_sdk import Wallet

# This example lists all accounts in the wallet.

# This example uses secrets in environment variables for simplicity which
# should not be done in production.
load_dotenv()

wallet = Wallet(os.environ['WALLET_DB_PATH'])

for account in wallet.get_accounts():
    print(account.get_metadata())

File: bindings/nodejs/examples/how_tos/accounts_and_addresses/list_accounts.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger } from '@iota/sdk';

// Run with command:
// yarn run-example ./how_tos/accounts_and_addresses/list_accounts.ts

// This example lists all accounts in the wallet
async function run() {
    initLogger();
    try {
        const wallet = new Wallet({
            storagePath: './alice-database',
        });

        const accounts = await wallet.getAccounts();

        for (const account of accounts)
            console.log(account.getMetadata().alias);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
